### DISCUSSION TITLE
Migration from StealthChestnut: a few queries

### DISCUSSION THREAD

**Discussion by C00K50N** [nathanmarlor/foxess_modbus#475] [Support] [answered]
I have finally taken the plunge and installed a Waveshare RS485 to Ethernet device in order to stop using the inverter's built in LAN connection.  So last night I followed the migration guide and appear to have got pretty much everything up and running using this integration.... I think!  But I have a few queries:

Do I need to remove the previous integration from HACS?  Or is it OK / better to leave it installed?

This doesn't replace @nathanmarlor's other integration FoxESS Energy Management, does it?  I still have that installed and I take it I need to keep it in order to have charging automated based on weather forecast, etc.  But I've changed my config on that so it's now accessing via RS485 rather than Cloud.

I had a gauge set up previously which swung left or right from the centre based on live charge / discharge.  It read from the discharge sensor, which if charging went negative.  The replacement sensors don't seem to read negative, so is there a way to replicate that gauge with the new sensors?

![image](https://github.com/nathanmarlor/foxess_modbus/assets/102669305/93aaa37d-92cd-41f2-bd11-8c02b47c3729)
(left hand gauge, I'd like it to swing left if charging and right if discharging)

**Comment by canton7** [comment, marked as answer]
> Do I need to remove the previous integration from HACS? Or is it OK / better to leave it installed?

Probably doesn't matter much either way. It's not hard to reinstall if you need. Just make sure it's not configured to try and make endless fruitless attempts to talk to an inverter which is no longer there!

> This doesn't replace @nathanmarlor's other integration FoxESS Energy Management, does it?

No, they're separate. There are plans to make EM use this integration for the modbus comms side of things, but no action so far...
 
> I had a gauge set up previously which swung left or right from the centre based on live charge / discharge. It read from the discharge sensor, which if charging went negative. The replacement sensors don't seem to read negative, so is there a way to replicate that gauge with the new sensors?

Use the "Inverter Battery Power" (sensor.invbatpower) sensor.

**Reply by C00K50N** [reply]
Amazing! Thank you so much.  Looks like Inverter Battery Power is exactly what I was after! Perfect.

Forgive my lack of knowledge in what's probably a simple thing, but how do I check if the previous integration is still trying to connect to the inverter?

**Reply by canton7** [reply]
As long as you removed this line from your config, you should be fine:

```
modbus: !include custom_components/foxmodbuslan/modbusUSB.yaml
```

**Reply by C00K50N** [reply]
I did indeed.  That's great, thank you again!

**Comment by ianhills1000** [comment]
Good afternoon I have seen your emails and really could do with some advice if possible?If I understand correctly you have linked your Fox ESS to Home Assistant. I am in the very beginning of looking at this and have not got passed the first stage of how to physically link the Fox ESS K series to my raspberry pi running HA.I have a Fox ESS inverter K series with 12.5 kw batteries which was recently installed. I have only just got HA working with all my Tuya devices so it fair to say I am a newbie to this so please forgive my ignorance.WSent from my iPhoneOn 16 Dec 2023, at 11:58, C00K50N ***@***.***> wrote:﻿
I did indeed.  That's great, thank you again!

**Comment by ianhills1000** [comment]
Sorry sent incomplete message by accident.I see you used a Waveshare RS485 (mobs) to Ethernet converter. Where did you actually connect your RS485 cable to in the inverter. On the batteries there is in my situation a spare RS485 RJ45 plug which I assume is daisy chained to the inverter.Would be helpful to know how physically this networking is achieved if possible.Many thanks Ian HillsSent from my iPhoneOn 16 Dec 2023, at 13:15, IAN HILLS ***@***.***> wrote:﻿Good afternoon I have seen your emails and really could do with some advice if possible?If I understand correctly you have linked your Fox ESS to Home Assistant. I am in the very beginning of looking at this and have not got passed the first stage of how to physically link the Fox ESS K series to my raspberry pi running HA.I have a Fox ESS inverter K series with 12.5 kw batteries which was recently installed. I have only just got HA working with all my Tuya devices so it fair to say I am a newbie to this so please forgive my ignorance.WSent from my iPhoneOn 16 Dec 2023, at 11:58, C00K50N ***@***.***> wrote:﻿
I did indeed.  That's great, thank you again!

**Reply by C00K50N** [reply]
Hi, I am by no means the authority on this!  I come here for advice!!  So best ask others who know more, but to give you a starting point the guides I followed are here: 

https://github.com/nathanmarlor/foxess_modbus/wiki/Modbus-Wiring-Guide
and
https://github.com/nathanmarlor/foxess_modbus/wiki/Waveshare-RS485-to-ETH-%28B%29

**Comment by ianhills1000** [comment]
Hi,

Yes, this is all new to me and finding my around. I found these excellent documents on Wiki in the links. Thanks answers most of my questions and hopefully now will be able to get the Fox ESS linked up.

 

Many thanks for your reply.

**Reply by FozzieUK** [reply]
@ianhills1000 On the batteries the RS485 you mentioned is to allow the batteries to communicate amongst themselves and the BMS, there is then a Canbus connection to the inverter, but the RS485 connection you actually want is on the inverter itself and comes out of the meter/ct connection.

If you follow the modbus wiring guide link that @C00K50N posted above it will explain where and how, the KH has the same connections and pinouts as for the H1 inverter.

**Comment by ianhills1000** [comment]
Brilliant thanks! Very helpful
